python/xorbits/_mars/services/task/analyzer/fusion.py

```diff
@@ -110,7 +113,15 @@ def color(self) -> Dict[ChunkType, int]:
         chunk_to_colors = dict()
 
         # step 1: Coloring the initial nodes according to the bands that assigned by assigner
-        op_to_colors = self._color_init_nodes()
+        # if the number of initial color is less than the number of bands,
+        # decrease initial_same_color_num until the number of initial color exceeds bands.
+        while self.initial_same_color_num > 0:
+            op_to_colors = self._color_init_nodes()
+            if len(set(op_to_colors.values())) >= len(self.all_bands):
```

Comment:
10 nodes with init color 2 generates 5 subtasks.
If I have 1 worker band with 10 slots, then 5 slots will be unused.
Maybe we should compare it to the num of all slots?

Reply:
Good idea.